hackathon/views.py

```diff
@@ -188,3 +190,46 @@ def delete_hackathon(request, hackathon_id):
 
     messages.success(request, f'{hackathon.display_name} has been successfully deleted!')
     return redirect("hackathon:hackathon-list")
+
+class HackathonDetailView(DetailView):
+    """Renders a page with Hackathon details."""
+    model = Hackathon
+    context_object_name = "hackathon"
+
+
+def ajax_enroll_toggle(request):
+    """Swaps between being enrolled as a judge and unenrolling."""
+    # Checks to make sure the user has the permissions to enroll
+    user = request.user
+    data = {}
+    if (request.method == "POST"):
+
+        # Gets the PK of the Hackathon and then the related Hackathon
+        hackathon_id = request.POST.get("hackathon-id")
+        hackathon = Hackathon.objects.get(pk=hackathon_id)
+
+        # Check if user is staff or participant
+        if user.is_staff:
+        # Either enrolls or unenrolls a user from the judges
+            if user in hackathon.judges.all():
+                hackathon.judges.remove(user)
+                data["message"] = "You have withdrawn from judging."
+            else:
+                hackathon.judges.add(user)
+                data["message"] = "You have enrolled as a judge."
+        
+        else:
+            '''Check if user is already enrolled
```

Comment:
You are mixing your comment format here. As mentioned above though, I think you can remove this comment as well.

Reply:
Removed all comments on this view